Source/Core/getStringFromTypedArray.js

```diff
@@ -0,0 +1,49 @@
+/*global define*/
+define([
+        './defined',
+        './DeveloperError'
+    ], function(
+        defined,
+        DeveloperError) {
+    "use strict";
+    /*global TextDecoder*/
+
+    /**
+     * @private
+     */
+    var getStringFromTypedArray = function(buffer, byteOffset, length) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(buffer)) {
+            throw new DeveloperError('buffer is required.');
+        }
+
+        if (!defined(byteOffset)) {
+            throw new DeveloperError('byteOffset is required.');
+        }
+
+        if (!defined(length)) {
+            throw new DeveloperError('length is required.');
+        }
+        //>>includeEnd('debug');
+
+        return getStringFromTypedArray.decode(new Uint8Array(buffer, byteOffset, length));
+    };
+
+    // Exposed functions for testing
+    getStringFromTypedArray.decodeWithTextDecoder = function(view) {
+        var decoder = new TextDecoder('utf-8');
+        return decoder.decode(view);
+    };
+
+    getStringFromTypedArray.decodeWithFromCharCode = function(view) {
+        return String.fromCharCode.apply(String, view);
```

Comment:
`fromCharCode` is not going to work in all cases ([details](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode)).  This is similar to our render labeling woes.  We might be able to use `fromCodePoint` and related ES6 functions (which can be polyfilled) to handle all cases.  Of course this is a private function and if it works in the use cases we are using it for, then I'm fine with adding a comment or writing up an issue so that we at least document the limitations.